config/initializers/zeitwerk.rb

```diff
@@ -9,3 +9,20 @@
 Rails.autoloaders.main.collapse(Rails.root.join("lib/manageiq/reporting/charting"))
 Rails.autoloaders.main.collapse(Rails.root.join("lib/ansible/runner/credential"))
 Rails.autoloaders.main.collapse(Rails.root.join("lib/pdf_generator"))
+
+DescendantLoader.instance.discovered_parent_child_classes.each do |parent, children|
+  next if parent == "ApplicationRecord"
+
+  # puts "Registering on_load for class: #{parent}"
+  Rails.autoloaders.main.on_load(parent.to_s) do |klass, _abspath|
+    # puts "Running on_load for class: #{klass} with children: #{children}"
+    children.each do |child|
+      begin
+        child.safe_constantize
+      rescue NameError => err
```

Comment:
Can safe_constantize actually raise a NameError?  I thought under the covers they caught that and just returned nil.

Reply:
yes, it actually did, thanks for reminding me.  I need to investigate why.

Reply:
So, the short of it is that some top level classes can introduce cycles if you try to constantize their children.  Their children can reference constants that get loaded and loading of mixins and perent classes and some of those kick off more constantizing of children classes and some of those can be cyclic loads. So far,`MiqDecorator`, `OrchestrationStack::Status`, and `ResourcePool` look to be the problems.

Reply:
Ewwww ResourcePool is bad.  We need to fix that

Reply:
yeah, I'm getting things green with the current code and will try to figure out how to fix these classes or fix how we load.